Libplanet.Store/Trie/IRecordableTrie.cs

```diff
@@ -0,0 +1,15 @@
+using Bencodex.Types;
+
+namespace Libplanet.Store.Trie
+{
+    public interface IRecordableTrie : ITrie
+    {
+         /// <summary>
+         /// Cleans up and stores the <see cref="IRecordableTrie"/> in storage.
+         /// </summary>
+         /// <returns>Returns new committed <see cref="IRecordableTrie"/>.</returns>
+        IRecordableTrie Commit();
+
+        new IRecordableTrie Set(in KeyBytes key, IValue value);
```

Comment:
Why does `IRecordableTrie` need `Set()`? Shouldn't `Set()` be done via `ITrie`? 🤔

Reply:
https://github.com/planetarium/libplanet/pull/3392/commits/7cc6de819734f5c25355005807a5c31ccd52025b for convenience

Reply:
Weighing the convenience gained against unnecessary cluttering of an `interface`, if it was up to me, I'd say it'd be better to get rid of the whole extensions. 🙄